proxy/http2/unit_tests/test_Http2Frame.cc

```diff
@@ -39,13 +39,13 @@ TEST_CASE("Http2Frame", "[http2][Http2Frame]")
     uint8_t hdr_block_len = sizeof(hdr_block);
 
     Http2PushPromiseFrame frame(id, flags, pp, hdr_block, hdr_block_len);
-    uint64_t written = frame.write_to(miob);
+    int64_t written = frame.write_to(miob);
 
     CHECK(written == HTTP2_FRAME_HEADER_LEN + sizeof(Http2StreamId) + hdr_block_len);
     CHECK(written == miob_r->read_avail());
 
     uint8_t buf[32] = {0};
-    uint64_t read   = miob_r->read(buf, written);
+    int64_t read    = miob_r->read(buf, written);
```

Comment:
Just a quick question on this on this looked odd. should `miob_r->read(buf, written);` return an unsigned?

or does it return a signed so you are changing uint64_t to init64_t to avoid a warning?

Reply:
I might have found the answer... it looks like we are having lots of size_t values as return values.. so you are getting a lot of these issues 

```
unit_tests/test_Http2Frame.cc:44:19: error: comparison of integer expressions of different signedness: 'int64_t' {aka 'long int'} and 'size_t' {aka 'long unsigned int'} [-Werror=sign-compare]
     CHECK(written == HTTP2_FRAME_HEADER_LEN + sizeof(Http2StreamId) + hdr_block_len);
```

Reply:
The answer is use `ssize_t`.